### ISSUE TITLE
Cycling through audio devices - notification delay

### ISSUE THREAD

**Issue by jhakonen** [Belphemur/SoundSwitch#163]
I have three playback devices which I'm switching through using SoundSwitch in Windows 10:
 * Kaiuttimet (Realtek High Definition Audio)
 * Kaiuttimet (Sound Blaster Z)
 * ROG PG348Q-8 (NVIDIA High Definition Audio)

When I want to listen audio e.g. from Spotify, I start the application and start playing some track. Then I press SoundSwitch's playback hotkey rapidly several times until audio comes out from correct audio device.

However it would be nice if I could select the device before starting audio playback.

SoundSwitch shows notifications when I cycle through the devices, but they pop up with ~10 second intervals, which is way too long time to wait between hotkey presses to see which device is currently selected.

How do I get rid of that 10 second interval? I would like to rapidly change the audio devices and still be able to see from the notification which device is currently selected.

**Comment by FireEmerald** [comment]
I would say the easiest way would be to use another Notification type. In your case a Sound Notification which will play a sound on the selected device.

It is not possible to set a delay below 5sec for the windows notifications. Options can be found here:
![show_notifications_for_in_settings](https://cloud.githubusercontent.com/assets/1107939/23827493/4c2bf94e-06b5-11e7-8a75-0d220d77731a.jpg)

> NOTE: The default time is 5 seconds. Entering anything lower then 5 seconds will still have a 5 second wait.
Source: [6175-notifications-display-time-change-window-10](https://www.tenforums.com/tutorials/6175-notifications-display-time-change-window-10-a.html#option3)

**Comment by jhakonen** [comment]
Thank you,

That sound notification works for me a lot better. It still would be nice to have visual confirmation, but this will do!

**Comment by Belphemur** [comment]
As @FireEmerald said this is an annoying restriction of windows 8/10. 

In the code of SoundSwitch I put a 2 second delay but as you can see it's not taken into account. 

To improve this, we'd have to create our own type of notification pop-up instead of relying on windows to do it. 

⁣Sent from BlueMail ​
